src/hooks.rs

```diff
@@ -10,6 +10,17 @@ use crate::hints::MOCK_CALL_KEY;
 
 pub const HOOKS_VAR_NAME: &str = "hooks";
 
+/// Called before an instruction is executed by the virtual machine (VM).
+/// 
+/// This function is called before the VM will execute an instruction.
+/// The given `&mut VirtualMachine` allow to modify VM state
```

Comment:
allow to modify VM state -> allows the VM state to be modified

Reply:
PR updated